docs/demo/demo.js

```diff
@@ -52,70 +52,133 @@ $clearElem.addEventListener('click', handleClearClick, false);
     return 'dark';
   }
-
   return 'light';
 }
 
-// Apply theme on page load
-const initialTheme = getPreferredTheme();
-applyTheme(initialTheme);
-
-if ($themeToggle) {
-  $themeToggle.addEventListener('click', function() {
-    const currentTheme = document.body.classList.contains('dark')
-      ? 'dark'
-      : 'light';
-    const newTheme = currentTheme === 'dark' ? 'light' : 'dark';
+function sanitisePreference(value) {
+  return value === 'dark' || value === 'light' || value === 'system' ? value : null;
+}
 
-    applyTheme(newTheme);
-    localStorage.setItem('theme', newTheme);
+function readStoredPreference() {
+  try {
+    const stored = sanitisePreference(localStorage.getItem(THEME_STORAGE_KEY));
+    if (stored) {
+      return stored;
+    }
+    return sanitisePreference(localStorage.getItem(LEGACY_STORAGE_KEY));
+  } catch {
+    return null;
+  }
+}
 
-    // Update iframe if it exists
-    try {
-      if ($previewIframe && $previewIframe.contentDocument) {
-        if (newTheme === 'dark') {
-          $previewIframe.contentDocument.documentElement.classList.add('dark');
-        } else {
-          $previewIframe.contentDocument.documentElement.classList.remove(
-            'dark',
-          );
-        }
-      }
-    } catch {
-      // Ignore cross-origin errors
+function writeStoredPreference(preference) {
+  try {
+    localStorage.setItem(THEME_STORAGE_KEY, preference);
+    if (preference === 'light' || preference === 'dark') {
+      localStorage.setItem(LEGACY_STORAGE_KEY, preference);
+    } else {
+      localStorage.removeItem(LEGACY_STORAGE_KEY);
     }
+  } catch {
+    // Storage might be unavailable; ignore
+  }
+}
+
+function getEffectiveTheme(preference) {
+  return preference === 'system' ? getSystemTheme() : preference;
+}
+
+function updateToggle(preference) {
+  if (!$themeToggle) {
+    return;
+  }
+  const details = TOGGLE_UI[preference] || TOGGLE_UI.system;
+  if ($themeToggleIcon) {
+    $themeToggleIcon.textContent = details.icon;
+  }
+  if ($themeToggleText) {
+    $themeToggleText.textContent = details.text;
+  }
+  $themeToggle.setAttribute('data-theme-mode', preference);
+  const label = `Switch theme (current: ${details.text})`;
+  $themeToggle.setAttribute('aria-label', label);
+  $themeToggle.title = label;
+}
+
+let currentPreference = readStoredPreference() || 'system';
+
+function applyPreference(preference, persist) {
+  currentPreference = preference;
+  const effectiveTheme = getEffectiveTheme(preference);
+  applyTheme(effectiveTheme);
+  document.documentElement.setAttribute('data-theme-preference', preference);
+  updateToggle(preference);
+  if (persist) {
+    writeStoredPreference(preference);
+  }
+}
+
+applyPreference(currentPreference, true);
+
+if ($themeToggle) {
+  $themeToggle.addEventListener('click', function() {
+    const index = THEME_ORDER.indexOf(currentPreference);
+    const nextIndex = index === -1 ? 0 : (index + 1) % THEME_ORDER.length;
```

Comment:
```suggestion
    const nextIndex = (index + 1) % THEME_ORDER.length;
```

(-1 + 1) is the same as `0`